[AIRFLOW-3104] Add .airflowignore info into doc

XD-DENG opened the pull request:
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3104
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

`.airflowignore` is a nice feature, but it was not mentioned at all anywhere in the documentation.

I added relevant information in `doc/concepts`.

Fixed minor typos BTW.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

kaxil commented on airflow/models.py:
```diff
@@ -522,10 +522,12 @@ def collect_dags(
-        Note that if a .airflowignore file is found while processing,
-        the directory, it will behaves much like a .gitignore does,
+        Note that if a .airflowignore file is found while processing
+        the directory, it will behave much like a .gitignore does,
```
either remove `does` or change it to `Note that if a ``.airflowignore`` file is found while processing
        the directory, it will behave much like how a ``.gitignore`` does, the directory, it will behave much like a .gitignore does`

XD-DENG replied:
Updated as suggested.

1 (out of 9) CI test failed due to irrelevant error.

r39132 commented on airflow/models.py:
```diff
@@ -522,10 +522,12 @@ def collect_dags(
 
-        Note that if a .airflowignore file is found while processing,
-        the directory, it will behaves much like a .gitignore does,
+        Note that if a ``.airflowignore`` file is found while processing
```
Where in the file system does this .airflowignore need to be? Why not have a placeholder (empty) .airflowignore committed or part of the distribution, so folks know to override it in place?

XD-DENG replied:
This .airflowignore file needs to be in DAG_FOLDER, then it can be detected when we parse the DAG files.

It’s a great idea to have a placeholder .airflowignore. But it’s a bit tricky to do this since users normally need to create “dags” folder by themselves after initdb.

XD-DENG replied:
I didn’t know the existence of this nice feature either previously. that’s exactly why we need to mention it to users more explicitly in the documentation ;-)

r39132 reviewed (changes requested):
See inline comments

ashb commented on docs/concepts.rst:
```diff
@@ -37,6 +37,12 @@ logical workflow.
+
+   If a ``.airflowignore`` file is found while processing ``DAG_FOLDER``,
+   it will behave much like a ``.gitignore``, ignoring files that match
+   any of the regex patterns specified in the file.
```
Should probably mention the format of the lines in the file, and perhaps give an example here? At that point this is possibly a (sub)section in its own right, and not a mere note.

XD-DENG replied:
Sure, it's worthwhile doing that. Where would you suggest to put this section? A section under `Concepts/Core Ideas/DAGs` or `Concepts`? Or somewhere else?

ashb replied:
Either a section under Concepts -> Additional Functionality, or directly under Concepts

XD-DENG replied:
It may make more sense to put it directly under `Concepts` to me.

Will modify accordingly and ping you then. Cheers

ashb commented on docs/concepts.rst:
```diff
@@ -863,3 +863,30 @@ do the same, but then it is more to use a virtualenv and pip.
+
+
+Then files like "project_a_dag_1.py", "TESTING_project_a.py", "tenant_1.py",
+"project_a/dag_1.py", and "tenant_1/dag_1.py" in your ``DAG_FOLDER`` would be ignored.
```
One small clarification here (that maybe doesn't matter) but if a directory matches a pattern then Airflow doesn't look at any files or sub-directories in it either - this is useful to exclude whole directories with a large number of files which can slow down airflow DAG finding.

XD-DENG replied:
Worth mentioning to users. Sorry I missed to cover this.

Let me modify later and ping you then.

XD-DENG committed:
[AIRFLOW-3104] Add .airflowignore info into doc

.airflowignore is a nice feature,
but it was not mentioned at all in the documentation.
